src/stores/courseData.js

```diff
@@ -0,0 +1,232 @@
+        changed: moment().format(DATE_FORMAT),
+        name: 'My First Road',
+        agent: '',
+        contents: {
+          coursesOfStudy: ['girs'],
+          selectedSubjects: [],
+          progressOverrides: {}
+        }
+      }
+    },
+    subjectsIndex: {},
+    subjectsInfo: []
+  },
+  mutations: {
+    addClass (state, newClass) {
+      state.roads[state.activeRoad].contents.selectedSubjects.push(newClass);
+      Vue.set(state.roads[state.activeRoad], 'changed', moment().format(DATE_FORMAT));
+    },
+    addReq (state, event) {
+      state.roads[state.activeRoad].contents.coursesOfStudy.push(event);
+      Vue.set(state.roads, state.activeRoad, state.roads[state.activeRoad]);
+    },
+    allowCookies (state) {
+      state.cookiesAllowed = true;
+    },
+    clearClassInfoStack (state) {
+      state.classInfoStack = [];
+    },
+    disallowCookies (state) {
+      state.cookiesAllowed = false;
+    },
+    deleteRoad (state, id) {
+      Vue.delete(state.roads, id);
+    },
+    moveClass (state, { classIndex, semester }) {
+      state.roads[state.activeRoad].contents.selectedSubjects[classIndex].semester = semester;
+      Vue.set(state.roads[state.activeRoad], 'changed', moment().format(DATE_FORMAT));
+    },
+    overrideWarnings (state, payload) {
+      const classIndex = state.roads[state.activeRoad].contents.selectedSubjects.indexOf(payload.classInfo);
+      Vue.set(state.roads[state.activeRoad].contents.selectedSubjects[classIndex], 'overrideWarnings', payload.override);
+    },
+    parseGenericCourses (state) {
+      const girAttributes = {
+        'PHY1': ['Physics 1 GIR', 'p1'],
+        'PHY2': ['Physics 2 GIR', 'p2'],
+        'CHEM': ['Chemistry GIR', 'c'],
+        'BIOL': ['Biology GIR', 'b'],
+        'CAL1': ['Calculus I GIR', 'm1'],
+        'CAL2': ['Calculus II GIR', 'm2'],
+        'LAB': ['Lab GIR', 'l1'],
+        'REST': ['REST GIR', 'r']
+      };
+      // the titles of the hass and ci attributes are currently not used in the description on fireroad
+      // I think they might be nice to display with the description, but as of now they are unused
+      const hassAttributes = {
+        'HASS-A': ['HASS Arts', 'ha'],
+        'HASS-S': ['HASS Social Sciences', 'hs'],
+        'HASS-H': ['Hass Humanities', 'hh']
+      };
+      const ciAttributes = {
+        'CI-H': ['Communication Intensive', 'hc'],
+        'CI-HW': ['Communication Intensive with Writing', 'hw']
+      };
+      const genericCourses = [];
+      const baseGeneric = {
+        description: 'Use this generic subject to indicate that you are fulfilling a requirement, but do not yet have a specific subject selected.',
+        total_units: 12
+      };
+      // biol:b, chem: c, lab: l1, partial lab: l2, rest: r, calc1: m1, calc2: m2, phys1: p1, phys2: p2
+      // hass-a: ha, hass-h: hh, hass-s: hs, hass elective: ht, hass subject: h%5Bahst%5D
+      // commun_int - cih: hc, cihw: hw
```

Comment:
Anyone else have comments on these comments? I want to get a second opinion before inadvertently deleting any important comments.

Reply:
I believe that was just for my reference as I was constructing the three attribute dicts.  The only thing that might be useful to note is the `hass subject: h%5Bahst%5D` part of the comment, since we don't have any generic HASS classes (just HASS-X), and this would be useful information if we wanted to do that in the future, but it shouldn't be left as a comment here.

Reply:
I'l remove the comments and move them to Mattermost or something until we find a better place to put it.